model-engine/model_engine_server/inference/vllm/vllm_server.py

```diff
@@ -38,7 +40,30 @@ async def generate(request: Request) -> Response:
     request_dict = await request.json()
     prompt = request_dict.pop("prompt")
     stream = request_dict.pop("stream", False)
+    guided_json = request_dict.pop("guided_json", None)
+    guided_regex = request_dict.pop("guided_regex", None)
+    guided_choice = request_dict.pop("guided_choice", None)
     sampling_params = SamplingParams(**request_dict)
+
+    # Dummy request to get guided decode logit processor
+    partial_openai_request = OpenAICompletionRequest.model_validate(
```

Comment:
a validation error here would be thrown as a 500 right? lets make sure its a 400 instead?

Reply:
sure. although validation failure should happen upstream at gateway